File: 2、LBOCExpandFriendList/LBExpandFriendList/QQ好友列表/LBQQFriendCell.m
//
//  LBQQFriendCell.m
//  LBExpandFriendList
//
//  Created by yunmei on 2018/1/20.
//  Copyright © 2018年 yunmei. All rights reserved.
//

#import "LBQQFriendCell.h"
#import "Masonry.h"
@implementation LBQQFriendCell{
    UIImageView *_imageView;
}
+ (instancetype)cellWithTableView:(UITableView *)tableview{
    
    LBQQFriendCell *cell = [tableview dequeueReusableCellWithIdentifier:@"LBQQFriendCell"];
    if (!cell) {
        cell = [[LBQQFriendCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"LBQQFriendCell"];
    }
    tableview.separatorStyle = UITableViewCellSeparatorStyleNone;
    //    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    return cell;
}
-(void)setDatadic:(NSDictionary *)datadic{
    _datadic = datadic;
    _titleLabel.text = _datadic[@"name"];
}
-(void)awakeFromNib{
    [super awakeFromNib];
}
-(instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        
        [self setUpLayout];
    }
    return  self;
}
- (void)setUpLayout{
    /// 图标
    _imageView = [[UIImageView alloc]init];
    _imageView.image = [UIImage imageNamed:@"icon.jpg"];
    _imageView.layer.cornerRadius = 25;
    _imageView.layer.masksToBounds = YES;
    [self.contentView addSubview:_imageView];
    __weak typeof(self.contentView) weakSelf = self.contentView;
    [_imageView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerY.equalTo(weakSelf);
        make.left.equalTo(weakSelf).offset(25);
        make.width.height.equalTo(@50);
    }];
    /// 标题
    _titleLabel = [[UILabel alloc]init];
    _titleLabel.font = [UIFont systemFontOfSize:14];
    [self.contentView addSubview:_titleLabel];
    [_titleLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.top.equalTo(weakSelf).offset(8);
        make.left.mas_equalTo(_imageView.mas_right).offset(10);
    }];
    /// 在线状态
    _onlineLabel = [[UILabel alloc]init];
    _onlineLabel.font = [UIFont systemFontOfSize:14];
    [self.contentView addSubview:_onlineLabel];
    [_onlineLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.mas_equalTo(_imageView.mas_bottom);
        make.left.mas_equalTo(_imageView.mas_right).offset(10);
    }];
    /// 昵称
    _nikeLabel = [[UILabel alloc]init];
    _nikeLabel.font = [UIFont systemFontOfSize:14];
    [self.contentView addSubview:_nikeLabel];
    [_nikeLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.mas_equalTo(_imageView.mas_bottom);
        make.left.mas_equalTo(_onlineLabel.mas_right).offset(10);
    }];
}

@end

File: LBSwiftExpandFriendList/LBSwiftExpandFriendList/QQ好友列表/LBQQFriendCell.swift
//
//  LBQQFriendCell.swift
//  LBSwiftExpandFriendList
//
//  Created by yunmei on 2018/1/22.
//  Copyright © 2018年 liubo. All rights reserved.
//

import UIKit

class LBQQFriendCell: UITableViewCell {

    lazy var titleLabel:UILabel = UILabel()
    lazy var onlineLabel:UILabel = UILabel()
    lazy var nikeLabel:UILabel = UILabel()
    lazy var imageV:UIImageView = UIImageView()
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        setUpLayout()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

}
extension LBQQFriendCell{
    func setUpLayout(){
        /// 图标
        imageV.image = UIImage(named: "icon.jpg")
        imageV.layer.cornerRadius = 25
        imageV.layer.masksToBounds = true
        contentView.addSubview(imageV)
        imageV.snp.makeConstraints { (make) in
            make.centerY.equalTo(contentView)
            make.left.equalTo(contentView).offset(25)
            make.width.height.equalTo(50)
        }
        /// 标题
        titleLabel.font = UIFont.systemFont(ofSize: 14)
        contentView.addSubview(titleLabel)
        titleLabel.snp.makeConstraints { (make) in
            make.top.equalTo(contentView).offset(8)
            make.left.equalTo(imageV.snp.right).offset(10)
        }
        /// 在线状态
        onlineLabel.font = UIFont.systemFont(ofSize: 14)
        contentView.addSubview(onlineLabel)
        onlineLabel.snp.makeConstraints { (make) in
            make.bottom.equalTo(imageV.snp.bottom)
            make.left.equalTo(imageV.snp.right).offset(10)
        }
        /// 昵称
        nikeLabel.font = UIFont.systemFont(ofSize: 14)
        contentView.addSubview(nikeLabel)
        nikeLabel.snp.makeConstraints { (make) in
            make.bottom.equalTo(imageV.snp.bottom);
            make.left.equalTo(onlineLabel.snp.right).offset(10);
        }
    }
}
